src/graph.rs

```diff
@@ -126,6 +151,71 @@ impl Graph {
       }
     }
   }
+
+  /// Returns the number of dimensions of the Tensor referenced by `output`.
+  ///
+  /// If the number of dimensions in the shape is unknown, returns -1.
+  ///
+  /// Returns an error if:
+  ///   * `output` is not in `graph`.
+  pub fn num_dims(&self, output: Output) -> Result<c_int> {
+    let status = Status::new();
+    unsafe {
+      let val = tf::TF_GraphGetTensorNumDims(
+        self.gimpl.inner,
+        output.to_c(),
+        status.inner);
+      if status.is_ok() {
+        Ok(val)
+      } else{
+        Err(status)
+      }
+    }
+  }
+
+  /// Returns the shape of the Tensor referenced by `output`.
+  ///
+  /// If the number of dimensions in the shape is unknown, the return value is
+  /// None. Otherwise, each element of `dims` will be set corresponding to the
+  /// size of the dimension. An unknown dimension is represented by `-1`.
+  ///
+  /// Returns an error if:
+  ///   * `output` is not in `graph`.
+  pub fn tensor_shape(&self, output: Output) -> Result<Option<Vec<i64>>> {
```

Comment:
So something roughly like `type TensorShape = Vec<Option<i64>>` and have `tensor_shape()` return `Result<Option<TensorShape>>`?